modules/core/04-channel/keeper/upgrade_test.go

```diff
@@ -1264,76 +1264,116 @@ func (suite *KeeperTestSuite) TestChanUpgradeCancel() {
-			types.AttributeKeyChannelID:             path.EndpointA.ChannelID,
-			types.AttributeCounterpartyPortID:       path.EndpointB.ChannelConfig.PortID,
-			types.AttributeCounterpartyChannelID:    path.EndpointB.ChannelID,
-			types.AttributeKeyUpgradeConnectionHops: proposedUpgrade.Fields.ConnectionHops[0],
-			types.AttributeKeyUpgradeVersion:        proposedUpgrade.Fields.Version,
-			types.AttributeKeyUpgradeOrdering:       proposedUpgrade.Fields.Ordering.String(),
-			types.AttributeKeyUpgradeSequence:       fmt.Sprintf("%d", channel.UpgradeSequence),
-		},
-		sdk.EventTypeMessage: {
-			sdk.AttributeKeyModule: types.AttributeValueCategory,
-		},
-	}
+				channel = path.EndpointA.GetChannel()
 
-	ibctesting.AssertEvents(&suite.Suite, expEvents, events)
+				// Verify that channel has been restored to previous state
+				suite.Require().Equal(types.OPEN, channel.State)
+				suite.Require().Equal(mock.Version, channel.Version)
+				suite.Require().Equal(errorReceipt.Sequence, channel.UpgradeSequence)
 
-	counterpartyUpgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
-	suite.Require().Equal(types.Upgrade{}, counterpartyUpgrade)
-	suite.Require().False(found)
+				// Assert that state stored for upgrade has been deleted
+				upgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
+				suite.Require().Equal(types.Upgrade{}, upgrade)
+				suite.Require().False(found)
+
+				// we need to find the event values from the proposed upgrade as the actual upgrade has been deleted.
+				proposedUpgrade := path.EndpointA.GetProposedUpgrade()
+				events := ctx.EventManager().Events().ToABCIEvents()
+				expEvents := ibctesting.EventsMap{
+					types.EventTypeChannelUpgradeCancel: {
+						types.AttributeKeyPortID:                path.EndpointA.ChannelConfig.PortID,
+						types.AttributeKeyChannelID:             path.EndpointA.ChannelID,
+						types.AttributeCounterpartyPortID:       path.EndpointB.ChannelConfig.PortID,
+						types.AttributeCounterpartyChannelID:    path.EndpointB.ChannelID,
+						types.AttributeKeyUpgradeConnectionHops: proposedUpgrade.Fields.ConnectionHops[0],
+						types.AttributeKeyUpgradeVersion:        proposedUpgrade.Fields.Version,
+						types.AttributeKeyUpgradeOrdering:       proposedUpgrade.Fields.Ordering.String(),
+						types.AttributeKeyUpgradeSequence:       fmt.Sprintf("%d", channel.UpgradeSequence),
+					},
+					sdk.EventTypeMessage: {
+						sdk.AttributeKeyModule: types.AttributeValueCategory,
+					},
+				}
+
+				ibctesting.AssertEvents(&suite.Suite, expEvents, events)
+
+				counterpartyUpgrade, found := suite.chainA.GetSimApp().IBCKeeper.ChannelKeeper.GetCounterpartyUpgrade(suite.chainA.GetContext(), path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID)
```

Comment:
would be nice to move this further up along with the check for the chains upgrade. We do also delete CounterpartyLastPacketSequence but, that might just be removed in #4307

Reply:
yep, that will be removed, no need to check it here